x-pack/plugins/security_solution/public/cloud_security_posture/components/csp_details/insights_tab_csp.tsx

```diff
@@ -5,19 +5,105 @@
+  };
+}
+interface CspFlyoutPanels extends FlyoutPanels {
+  left: CspFlyoutPanelProps;
+}
+
 export const InsightsTabCsp = memo(
   ({ name, fieldName }: { name: string; fieldName: 'host.name' | 'user.name' }) => {
+    const panels = useExpandableFlyoutState() as CspFlyoutPanels;
+    const defaultTab = () =>
+      panels.left?.params?.hasMisconfigurationFindings
+        ? 'misconfigurationTabId'
+        : panels.left?.params?.hasVulnerabilitiesFindings
+        ? 'vulnerabilitiesTabId'
+        : 'undefined';
+
+    const [activeInsightsId, setActiveInsightsId] = useState(
+      panels.left?.params?.path?.subTab || defaultTab
+    );
+    const insightsButtonsNew: EuiButtonGroupOptionProps[] = useMemo(() => {
+      const buttons: EuiButtonGroupOptionProps[] = [];
+      if (panels.left?.params?.hasMisconfigurationFindings) {
+        buttons.push({
+          id: 'misconfigurationTabId',
+          label: (
+            <FormattedMessage
+              id="xpack.securitySolution.flyout.left.insights.misconfigurationButtonLabel"
+              defaultMessage="Misconfiguration"
+            />
+          ),
+          'data-test-subj': 'misconfigurationTabDataTestId',
+        });
+      }
+      if (panels.left?.params?.hasVulnerabilitiesFindings) {
+        buttons.push({
+          id: 'vulnerabilitiesTabId',
+          label: (
+            <FormattedMessage
+              id="xpack.securitySolution.flyout.left.insights.vulnerabilitiesButtonLabel"
+              defaultMessage="Vulnerabilities"
+            />
+          ),
+          'data-test-subj': 'vulnerabilitiesTabDataTestId',
+        });
+      }
+      return buttons;
+    }, [
+      panels.left?.params?.hasMisconfigurationFindings,
+      panels.left?.params?.hasVulnerabilitiesFindings,
+    ]);
+    const onTabChange = (id: string) => {
+      setActiveInsightsId(id);
+    };
+
     return (
       <>
+        <EuiButtonGroup
+          color="primary"
+          legend={i18n.translate(
+            'xpack.securitySolution.flyout.left.insights.optionsButtonGroups',
+            {
+              defaultMessage: 'Insights options',
+            }
+          )}
+          options={insightsButtonsNew}
+          idSelected={activeInsightsId}
+          onChange={onTabChange}
+          buttonSize="compressed"
+          isFullWidth
+          data-test-subj={'insightButtonGroupsTestId'}
+        />
         <EuiSpacer size="xl" />
-        <MisconfigurationFindingsDetailsTable fieldName={fieldName} queryName={name} />
+        {/* <MisconfigurationFindingsDetailsTable fieldName={fieldName} queryName={name} /> */}
+        {activeInsightsId === 'misconfigurationTabId' ? (
+          <MisconfigurationFindingsDetailsTable fieldName={fieldName} queryName={name} />
+        ) : (
+          // Render the vulnerabilities details table here when the vulnerabilities tab is selected
+          <VulnerabilitiesFindingsDetailsTable queryName={name} />
+        )}
```

Comment:
note that this comment is not 100% accurate since it looks like this table will be the default view. meaning if the id is not specifically set to `misconfigurationTabId`, the table will be `VulnerabilitiesFindingsDetailsTable`. for cases of `undefined` it can be a problem